src/core/xccl_tasks_queue.h

```diff
@@ -0,0 +1,26 @@
+/*
+* Copyright (C) Mellanox Technologies Ltd. 2001-2020.  ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#ifndef TASKS_QUEUE_H_
+#define TASKS_QUEUE_H_
+
+#include <malloc.h>
+#include "xccl_schedule.h"
+#include "xccl_progress_queue.h"
+
+typedef struct xccl_tasks_queue {
+    ucc_coll_task_t linked_list_head;
+    ucc_coll_task_t linked_list_tail;
+} xccl_tasks_queue_t;
```

Comment:
@vspetrov You asked me to use UCS list for the 1-threaded case. Because of different operations UCS list supports, I think it’s not as effective as possible. Please take a look at the file I used eventually, and let me know if you want it changed – maybe to some other UCS data structure?
In what I eventually used, Delete takes 1 op instead of 2, Insert takes 3 operations instead of 4. Because we said that one-threaded case is the most common, I guess every op counts, especially the delete and insert that will be called many times.